Initial InputEvent set of unit tests

IanGaither opened the pull request:
Covers all branches of non-virtual functions for all derivatives of the InputEvent class.

I'm not sure how to add the discussion label, but I was hoping to get some feedback on how to approach the rest of the class. Currently I've covered all the non-virtual functions (since it's abstract that required testing all the child classes). I think the rest of the functionality should be covered in tests that target each child directly, but I'm not sure as I've haven't worked with unit tests a ton and especially haven't worked with them in C++ before.

Any feedback would be appreciated!

Tests for class based on suggestions from: https://github.com/godotengine/godot/issues/43440#issue-740017075

IanGaither committed:
Initial InputEvent set of unit tests

Covers all branches of non-virtual functions for all derivatives of the InputEvent class.

IanGaither committed:
Replaced spaces with tabs and added class specifiers to the non virtual function calls

Cleaned up formatting issues and added class specifiers to hopefully resolve the macOS build issues.

akien-mga commented:
Seems to crash:
```
./tests/core/input/test_input_event.h:110:
TEST CASE:  [InputEvent] actions<InputEventWithModifiers>
  Action is present

./tests/core/input/test_input_event.h:110: FATAL ERROR: test case CRASHED: SIGABRT - Abort (abnormal termination) signal

/Users/runner/work/_temp/3aea7a72-8c05-4ea6-8c7f-a4eef80703f6.sh: line 3:  7549 Abort trap: 6           ./bin/godot.macos.editor.x86_64 --test
```

There are also style issues to fix, see the diff (wrong filename in header, and wrong indentation style): https://github.com/godotengine/godot/actions/runs/3467850975/jobs/5793096042

IanGaither committed:
Merge remote-tracking branch 'upstream/master' into input-event-unit-tests

IanGaither committed:
Replaced std tuple with macro implementation, added class type to subcases

IanGaither committed:
Removed tuple comment, fixed spacing in macro

Removed extra space

fixed spacing on multiline macro

removed tabs from spacing on multiline macro

Update test_input_event.h

Maran23 commented:
You could also add tests for `action_add_event` to add inputs to the action.

Calinou commented:
Note that https://github.com/godotengine/godot/pull/79444 has already added tests for InputEvent with a different, incompatible code structure.

It's still possible to add more unit tests to the now-existing file, but you'll need to refactor the code added by your pull request to do so.
